File: client/src/types/team_member.ts
export type TeamMemberSlot = 1 | 2 | 3 | 4 | 5;

export interface TeamMember {
  slot: TeamMemberSlot;
  name: string;
  designation: string;
  photoUrl: string;
  updatedAt: string;
}

export interface UpdateTeamMemberPayload {
  name: string;
  designation: string;
  photoBase64: string;
}

File: api/internal/models/team_member.go
package models

import "time"

type TeamMember struct {
	Slot         int       `gorm:"primaryKey"              json:"slot"`
	Name         string    `gorm:"size:255;not null;default:''" json:"name"`
	Designation  string    `gorm:"size:255;not null;default:''" json:"designation"`
	PhotoURL     string    `gorm:"column:photo_url"        json:"photoUrl"`
	CloudinaryID string    `gorm:"column:cloudinary_id;size:500" json:"-"`
	UpdatedAt    time.Time `gorm:"not null;default:now()"  json:"updatedAt"`
}
